src/language-selector/LanguageSelector.jsx

```diff
@@ -0,0 +1,118 @@
+import PropTypes from 'prop-types';
+import React, { useMemo, useContext } from 'react';
+import { getPrimaryLanguageSubtag, injectIntl } from '@edx/frontend-platform/i18n';
+import { Dropdown, useWindowSize } from '@openedx/paragon';
+import { Language } from '@openedx/paragon/icons';
+import { getCookies } from '@edx/frontend-platform/i18n/lib';
+import { AppContext } from '@edx/frontend-platform/react';
+
+/**
+ * Changes the user's language preference by setting a cookie and reloading the page.
+ *
+ * @param {string} langCookieName - The name of the cookie used to store language preference
+ * @param {string} selectedLocale - The locale code (e.g. 'en', 'es') to set as the preferred language
+ */
+const onLanguageSelected = (langCookieName, selectedLocale) => {
+  const cookies = getCookies();
+  cookies.set(langCookieName, selectedLocale);
+  window.location.reload();
+};
+
+/**
+ * Gets the localized display name of a language in its own language.
+ * First letter is capitalized for display purposes.
+ *
+ * @param {string} locale - The locale code (e.g. 'en', 'es')
+ * @returns {string} The capitalized display name of the language
+ */
+const getLocaleName = (locale) => {
+  const langName = new Intl.DisplayNames([locale], { type: 'language', languageDisplay: 'standard' }).of(locale);
+  return langName.charAt(0).toUpperCase() + langName.slice(1);
+};
+
+/**
+ * Language Selector component that displays a dropdown allowing users to change the site language.
+ *
+ * Features:
+ * - Responsive design that adjusts label display based on screen width
+ * - Only displays languages configured in SITE_SUPPORTED_LANGUAGES
+ * - Can be completely disabled via ENABLE_HEADER_LANG_SELECTOR config
+ * - Stores language preference in a cookie
+ *
+ * @component
+ * @param {Object} props
+ * @param {string} props.className - Additional CSS class names to apply to the component
+ * @returns {React.Element|null} The rendered component or null if disabled/no supported languages
+ */
+const LanguageSelector = ({ className }) => {
+  const { config } = useContext(AppContext);
+  const { width } = useWindowSize();
+  const cookies = getCookies();
+
+  const options = config.SITE_SUPPORTED_LANGUAGES;
```

Comment:
```suggestion
  const languageOptions = config.SITE_SUPPORTED_LANGUAGES;
```
I think options is to general